### ISSUE TITLE
Intermittent failure in ConcurrentTransactionTests on windows

### ISSUE THREAD

**Issue by benas** [spring-projects/spring-batch#3851]
`org.springframework.batch.core.test.concurrent.ConcurrentTransactionTests.testConcurrentLongRunningJobExecutions` fails intermittently on windows from the command line with the following error:

```
error	11-Feb-2021 08:38:26	[main] INFO org.springframework.batch.core.job.SimpleStepHandler - Executing step: [gameLoad]
build	11-Feb-2021 08:38:26	[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.078 s <<< FAILURE! - in org.springframework.batch.core.test.concurrent.ConcurrentTransactionTests
build	11-Feb-2021 08:38:26	[ERROR] testConcurrentLongRunningJobExecutions(org.springframework.batch.core.test.concurrent.ConcurrentTransactionTests)  Time elapsed: 0.063 s  <<< FAILURE!
build	11-Feb-2021 08:38:26	java.lang.AssertionError: expected:<FAILED> but was:<COMPLETED>
build	11-Feb-2021 08:38:26		at org.springframework.batch.core.test.concurrent.ConcurrentTransactionTests.testConcurrentLongRunningJobExecutions(ConcurrentTransactionTests.java:83)
build	11-Feb-2021 08:38:26	
build	11-Feb-2021 08:38:26	[INFO] Running org.springframework.batch.core.test.football.FootballJobIntegrationTests
```

**Comment by nfjcore** [comment]
This test also fails for me when running all tests in spring-batch-core.  The test passes in isolation. I'm on MacOS. Let me know if I can provide additional information.

**Comment by benas** [comment]
@nfjcore I'm also on MacOS (Catalina v10.15.7) and this test didn't fail up until now when I build locally with Java 8:

```
$ java -version
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)
```

It did not fail neither on our [CI server](https://build.spring.io/browse/BATCH-MCI-17). But if you say it fails on your machine, I trust you. 

> Let me know if I can provide additional information.

Yes, please. Which java version do you use? Do you use the maven wrapper or your local maven installation?

**Comment by nfjcore** [comment]
I am using the following java version:

```
java -version
openjdk version "1.8.0_282"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_282-b08)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.282-b08, mixed mode)
```

I should have specified it before, but ``org.springframework.batch.core.test.concurrent.ConcurrentTransactionTests`` fails only when I right click ``spring-batch-core/src/test/java`` and select ``Run 'All Tests'`` in Intellij. 

If I'm using the Maven wrapper ``LdifReaderTests.testValidRun`` is failing. But I guess this is beyond the scope of this issue.

**Comment by benas** [comment]
Seems to be resolved with #3856 (5 consecutive builds without failure).
